packages/strapi/lib/core/bootstrap.js

```diff
@@ -75,17 +75,10 @@ module.exports = function(strapi) {
   }, {});
 
   // Set components
-  strapi.components = Object.keys(strapi.components).reduce((acc, componentName) => {
+  Object.keys(strapi.components).forEach(componentName => {
     const component = strapi.components[componentName];
-    return {
-      ...acc,
-      [componentName]: _.merge(
-        // Ensure each component has a valid connection
-        { connection: component.connection || defaultConnection },
-        component
-      ),
-    };
-  }, {});
+    strapi.components[componentName].connection = component.connection || defaultConnection;
```

Comment:
you can do component.connection = ... ;)

Reply:
8a30673334c6755bf1b9fb4d44c2c937b3a16d56